frontend/src/component/pages/shared/CreateShortLinkSection.tsx

```diff
@@ -7,77 +7,188 @@ import { ShortLinkUsage } from './ShortLinkUsage';
-    if (!this.shortLinkTextField.current) {
-      return;
-    }
-    this.shortLinkTextField.current.focus();
-  };
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      longLink: '',
+      inputError: null,
+      shouldShowUsage: false,
+      createdShortLink: '',
+      createdLongLink: '',
+      qrCodeURL: ''
+    };
+  }
 
   render(): ReactElement {
     return (
       <Section title={'New Short Link'}>
         <div className={'control create-short-link'}>
           <div className={'text-field-wrapper'}>
             <TextField
-              defaultText={this.props.longLinkText}
+              ref={this.longLinkTextField}
+              defaultText={''}
               placeHolder={'Long Link'}
-              onBlur={this.props.onLongLinkTextFieldBlur}
-              onChange={this.props.onLongLinkTextFieldChange}
+              onBlur={this.handleLongLinkTextFieldBlur}
+              onChange={this.handleLongLinkChange}
             />
           </div>
           <div className={'text-field-wrapper'}>
             <TextField
               ref={this.shortLinkTextField}
-              defaultText={this.props.alias}
+              defaultText={''}
               placeHolder={'Custom Short Link ( Optional )'}
-              onBlur={this.props.onShortLinkTextFieldBlur}
-              onChange={this.props.onShortLinkTextFieldChange}
+              onBlur={this.handleCustomAliasTextFieldBlur}
+              onChange={this.handleAliasChange}
             />
           </div>
           <div className="create-short-link-btn">
-            <Button onClick={this.props.onCreateShortLinkButtonClick}>
+            <Button onClick={this.handleCreateShortLinkClick}>
               Create Short Link
             </Button>
           </div>
         </div>
-        <div className={'input-error'}>{this.props.inputErr}</div>
+        <div className={'input-error'}>{this.state.inputError}</div>
         {this.props.uiFactory.createPreferenceTogglesSubSection({
           uiFactory: this.props.uiFactory,
-          isShortLinkPublic: this.props.isShortLinkPublic,
-          onPublicToggleClick: this.props.onPublicToggleClick
+          isShortLinkPublic: this.state.isShortLinkPublic,
+          onPublicToggleClick: this.handlePublicToggleClick
         })}
-        {this.props.createdUrl && (
+        {this.state.shouldShowUsage && (
           <div className={'short-link-usage-wrapper'}>
             <ShortLinkUsage
-              shortLink={this.props.shortLink!}
-              originalUrl={this.props.createdUrl.originalUrl!}
-              qrCodeUrl={this.props.qrCodeUrl!}
+              shortLink={this.state.createdShortLink}
+              originalUrl={this.state.createdLongLink}
+              qrCodeUrl={this.state.qrCodeURL}
             />
           </div>
         )}
       </Section>
     );
   }
+
+  autoFillInLongLink(longLink: string) {
+    if (validateLongLinkFormat(longLink) != null) {
+      return;
+    }
```

Comment:
Is this needed here? We can just show the error when the user blurs out of the `InputField`.
For instance, if the long link is too long, the user has the option to make it smaller by removing any unwanted fragments(hashed fragments maybe) in the link.

Reply:
Okay, in that case, can we raise input error here instead of dropping it altogether?

Reply:
@rohithbalaji123 Sounds good. Will do.